### ISSUE TITLE
GPT2ForSequenceClassification fails accuracy in cpu inference after HF version upgrade

### ISSUE THREAD

**Issue by desertfire** [pytorch/pytorch#109019]
After upgrading the HF version used in CI (https://github.com/pytorch/pytorch/pull/107400), we see `fail_accuracy` for `GPT2ForSequenceClassification` when running cpu test, https://github.com/pytorch/pytorch/actions/runs/6133028313/job/16644698906

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10 @ezyang @msaroufim @wconstab @bdhirsh @anijain2305

**Comment by leslie-fang-intel** [comment]
Test on my local system, it looks like pass the accuracy check:
```
(inductor) [root@CPX-4 pytorch]# TRANSFORMERS_OFFLINE=1 OMP_NUM_THREADS=1 numactl -C 0-0 --membind=0 python benchmarks/dynamo/huggingface.py --accuracy --float32 -dcpu -n50 --no-skip --dashboard --batch-size 1 --threads 1 --only GPT2ForSequenceClassification --inference --freezing --timeout 9000 --backend=inductor --output=/tmp/inductor_single_test_st.csv
loading model: 0it [00:05, ?it/s]
cpu  eval  GPT2ForSequenceClassification
[2023-09-13 17:21:39,117] [2/0] torch._inductor.fx_passes.split_cat: [WARNING] example value absent for node: split_11
[2023-09-13 17:21:39,117] [2/0] torch._inductor.fx_passes.split_cat: [WARNING] example value absent for node: split_10
[2023-09-13 17:21:39,117] [2/0] torch._inductor.fx_passes.split_cat: [WARNING] example value absent for node: split_9
[2023-09-13 17:21:39,117] [2/0] torch._inductor.fx_passes.split_cat: [WARNING] example value absent for node: split_8
[2023-09-13 17:21:39,117] [2/0] torch._inductor.fx_passes.split_cat: [WARNING] example value absent for node: split_7
[2023-09-13 17:21:39,118] [2/0] torch._inductor.fx_passes.split_cat: [WARNING] example value absent for node: split_6
[2023-09-13 17:21:39,118] [2/0] torch._inductor.fx_passes.split_cat: [WARNING] example value absent for node: split_5
[2023-09-13 17:21:39,118] [2/0] torch._inductor.fx_passes.split_cat: [WARNING] example value absent for node: split_4
[2023-09-13 17:21:39,118] [2/0] torch._inductor.fx_passes.split_cat: [WARNING] example value absent for node: split_3
[2023-09-13 17:21:39,118] [2/0] torch._inductor.fx_passes.split_cat: [WARNING] example value absent for node: split_2
[2023-09-13 17:21:39,118] [2/0] torch._inductor.fx_passes.split_cat: [WARNING] example value absent for node: split_1
[2023-09-13 17:21:39,118] [2/0] torch._inductor.fx_passes.split_cat: [WARNING] example value absent for node: split
pass
```
I am using 
```
torch                       2.2.0a0+git41bd0fd
transformers                4.30.2
```

May I know should I upgrade `transformers` to some specific commit to reproduce this failure?

**Comment by jgong5** [comment]
@desertfire any comments on how to reproduce the failures?

**Comment by desertfire** [comment]
```
pip install "git+https://github.com/huggingface/transformers@`cat .ci/docker/ci_commit_pins/huggingface.txt`"
python benchmarks/dynamo/huggingface.py --ci --accuracy --inductor --device cpu --inference --float32 --only GPT2ForSequenceClassification
```

**Comment by leslie-fang-intel** [comment]
Thanks for the help. I thin I can reproduce this failure now

- CMD: `TRANSFORMERS_OFFLINE=1 OMP_NUM_THREADS=1 numactl -C 0-0 --membind=0 python benchmarks/dynamo/huggingface.py --accuracy --float32 -dcpu -n50 --no-skip --dashboard --batch-size 1 --threads 1 --only GPT2ForSequenceClassification --inference --freezing --timeout 9000 --backend=inductor --output=/tmp/inductor_single_test_st.csv`

- When use `transformer v4.30.2`, it can pass the accuracy check
- When use `transformer 6c26faa159b79a42d7fa46cb66e2d21523351987`, the accuracy check will fail.

**Comment by leslie-fang-intel** [comment]
Further investigation seems this transformer commit: https://github.com/huggingface/transformers/commit/f10452271802573fe6e19442631113c4c23a2c70 (PR: https://github.com/huggingface/transformers/pull/24979) makes some change to model script which causes this failure.

**Comment by leslie-fang-intel** [comment]
The root cause for this issue is after the model code changed, there are negative value tensor([-1]) used in another tensor's index.

- Here is a example to reproduce this failure:

```
logits = torch.randn((1, 4, 3)) # Failed

def test_fn2(primals_1, primals_2):
    full_default = torch.ops.aten.full.default([1], 0, dtype = torch.int64, layout = torch.strided, device = device(type='cpu'), pin_memory = False)
    index = torch.ops.aten.index.Tensor(primals_1, [full_default, primals_2]);  primals_1 = None
    return index
sequence_lengths = torch.tensor([-1])
correct3 = test_fn2(logits, sequence_lengths)
compiled3 = torch._dynamo.optimize_assert(compile_fx)(test_fn2)(logits, sequence_lengths)
print("correct3 is: {}".format(correct3), flush=True)
print("compiled3 is: {}".format(compiled3), flush=True)
``` 

- Draft PR: https://github.com/pytorch/pytorch/pull/111118 to fix this issue.

**Comment by leslie-fang-intel** [comment]
Discussed with @lezcano. Close PR https://github.com/pytorch/pytorch/pull/111118. And PR https://github.com/pytorch/pytorch/pull/108690 is expected to fix this issue also.

**Comment by leslie-fang-intel** [comment]
https://github.com/pytorch/pytorch/pull/108690 landed and verified with latest PyTorch master: 4f42edfb6e5b703eec2a14b8933090646702c5a2
```
pip install "git+https://github.com/huggingface/transformers@`cat .ci/docker/ci_commit_pins/huggingface.txt`"
python benchmarks/dynamo/huggingface.py --ci --accuracy --inductor --device cpu --inference --float32 --only GPT2ForSequenceClassification
```
@desertfire may I know is there any thing else we need to follow up to close this issue?

**Comment by lezcano** [comment]
I wonder why didn't CI go from failing accuracy to passing when submitting that PR...

**Comment by desertfire** [comment]
> #108690 landed and verified with latest PyTorch master: [4f42edf](https://github.com/pytorch/pytorch/commit/4f42edfb6e5b703eec2a14b8933090646702c5a2)
> 
> ```
> pip install "git+https://github.com/huggingface/transformers@`cat .ci/docker/ci_commit_pins/huggingface.txt`"
> python benchmarks/dynamo/huggingface.py --ci --accuracy --inductor --device cpu --inference --float32 --only GPT2ForSequenceClassification
> ```
> 
> @desertfire may I know is there any thing else we need to follow up to close this issue?

Let's take this test out of the CI skip list, https://github.com/pytorch/pytorch/blob/247f39f6039deffebb49b3fbe41b29d9e3a6bfb4/benchmarks/dynamo/common.py#L250

**Comment by leslie-fang-intel** [comment]
> > #108690 landed and verified with latest PyTorch master: [4f42edf](https://github.com/pytorch/pytorch/commit/4f42edfb6e5b703eec2a14b8933090646702c5a2)
> > ```
> > pip install "git+https://github.com/huggingface/transformers@`cat .ci/docker/ci_commit_pins/huggingface.txt`"
> > python benchmarks/dynamo/huggingface.py --ci --accuracy --inductor --device cpu --inference --float32 --only GPT2ForSequenceClassification
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > @desertfire may I know is there any thing else we need to follow up to close this issue?
> 
> Let's take this test out of the CI skip list,
> 
> https://github.com/pytorch/pytorch/blob/247f39f6039deffebb49b3fbe41b29d9e3a6bfb4/benchmarks/dynamo/common.py#L250

PR: https://github.com/pytorch/pytorch/pull/112100 to remove it from CI skip list, please help to take a look @desertfire

**Comment by leslie-fang-intel** [comment]
https://github.com/pytorch/pytorch/pull/112100 merged to take `GPT2ForSequenceClassification` out of the CI skip list
